builtin/gc.c

```diff
@@ -1615,15 +1615,46 @@ static int launchctl_remove_plists(const char *cmd)
 		launchctl_remove_plist(SCHEDULE_WEEKLY, cmd);
 }
 
+static int launchctl_list_contains_plist(const char *name, const char *cmd)
+{
+	int result;
+	struct child_process child = CHILD_PROCESS_INIT;
+	char *uid = launchctl_get_uid();
+
+	strvec_split(&child.args, cmd);
+	strvec_pushl(&child.args, "list", name, NULL);
+
+	child.no_stderr = 1;
+	child.no_stdout = 1;
+
+	if (start_command(&child))
+		die(_("failed to start launchctl"));
+
+	result = finish_command(&child);
+
+	free(uid);
+
+	/* Returns failure if 'name' doesn't exist. */
+	return !result;
+}
+
 static int launchctl_schedule_plist(const char *exec_path, enum schedule_priority schedule, const char *cmd)
 {
 	FILE *plist;
+	struct stat st;
 	int i;
 	const char *preamble, *repeat;
 	const char *frequency = get_frequency(schedule);
 	char *name = launchctl_service_name(frequency);
 	char *filename = launchctl_service_filename(name);
 
+	/* Does this file already exist? */
+	if (!stat(filename, &st) && st.st_size) {
+		/* Check if it is registered */
+		if (launchctl_list_contains_plist(name, cmd))
+			goto cleanup;
```

Comment:
Should we first verify that the contents are as expected? Otherwise, an updated `git maintenance` (that wants to write a slightly changed `.plist`) might not work as expected if the task had been registered previously?

Reply:
I implemented this.